encryptor/config/schemaStore.go

```diff
@@ -22,11 +22,19 @@ import (
 
 // TableSchemaStore fetches schema for encryptable tables in the database.
 type TableSchemaStore interface {
+	GetDatabaseConfig() *DatabaseConfig
 	// GetTableSchema returns schema for given table if configured, or nil otherwise.
 	GetTableSchema(tableName string) TableSchema
 	GetGlobalSettingsMask() SettingMask
 }
 
+// databaseConfig stores database-specific configuration that can affect connection
+// to the database, how SQL queries are processed and so on
+type DatabaseConfig struct {
```

Comment:
With getters, this thing does not need to be public. But there are also some tests in `decryptor/postgresql/proxy_test.go`, since there is a mock implementation of `TableSchemaStore`, and since we need to implement `GetDatabaseConfig() *DatabaseConfig` as part of that interface, we need `DatabaseConfig` to be available there, outside of `config` package.

Reply:
to implement `GetDatabaseConfig() DatabaseConfig`. Where `DatabaseConfig` is interface, not pointer to struct. And yes, in tests we should use some public implementations or mocks if we test other things